packages/broker/src/plugins/operator/InspectRandomNodeHelper.ts

```diff
@@ -0,0 +1,116 @@
+import { Contract } from "@ethersproject/contracts"
+import { operatorABI } from "@streamr/network-contracts"
+import type { Operator } from "@streamr/network-contracts"
+import { EthereumAddress, Logger, TheGraphClient, toEthereumAddress } from "@streamr/utils"
+import { OperatorServiceConfig } from "./OperatorPlugin"
+import fetch from 'node-fetch'
+
+const logger = new Logger(module)
+
+export class InspectRandomNodeHelper {
+
+    private readonly operatorContractAddress: EthereumAddress
+    private readonly operatorContract: Operator
+    private readonly theGraphClient: TheGraphClient
+
+    constructor(config: OperatorServiceConfig) {
+        this.operatorContractAddress = config.operatorContractAddress
+        const signer = config.signer.connect(config.provider)
+        this.operatorContract = new Contract(config.operatorContractAddress, operatorABI, signer) as unknown as Operator
+        this.theGraphClient = new TheGraphClient({
+            serverUrl: config.theGraphUrl,
+            fetch,
+            logger
+        })
+    }
+
+    async getSponsorshipsOfOperator(operatorAddress: EthereumAddress, requiredBlockNumber: number): Promise<EthereumAddress[]> {
+        
+        const createQuery = (lastId: string, pageSize: number) => {
+            return {
+                query: `
+                    {
+                        operator(id: "${operatorAddress}") {
+                            stakes(where: {id_gt: "${lastId}"}, first: ${pageSize}) {
+                                id
+                                sponsorship {
+                                    id
+                                }
+                            }
+                        }
+                        _meta {
+                            block {
+                            number
+                            }
+                        }
+                    }
+                    `
+            }
+        }
+        const parseItems = (response: { operator: { stakes: { id: string, sponsorship: { id: string } }[] } }): 
+        { id: string, sponsorship: { id: string } }[] => {
+            if (!response.operator) {
+                logger.error('Unable to find operator in The Graph', { operatorContractAddress: this.operatorContractAddress })
```

Comment:
Maybe it doesn't need to log anything? An empty list is a valid (and sometimes the expected) answer to the getSponsorshipsOfOperator() call.

Reply:
If that is the case, yeah then no logging required.

Reply:
I'll remove it, since the method is somewhat general, but the usecase "get sponsorships I am staked in" should maybe throw or log something, because if my own operatorcontract is not found something definitely is not right